build failure: pybuild installs during build_extension

Just updating the Debian packaging to 1.3.1 results in a build/install failure:

```
[100%] Built target gg
make[3]: Leaving directory '/<<PKGBUILDDIR>>/build/temp.linux-amd64-3.7'
Install the project...
-- Install configuration: "Release"
-- Installing: /usr/lib/python3/dist-packages/gau2grid
CMake Error at cmake_install.cmake:41 (file):
  file INSTALL cannot make directory
  "/usr/lib/python3/dist-packages/gau2grid": Permission denied


make[2]: *** [Makefile:118: install] Error 1
make[2]: Leaving directory '/<<PKGBUILDDIR>>/build/temp.linux-amd64-3.7'
Traceback (most recent call last):
  File "setup.py", line 147, in <module>
    zip_safe=False,
  File "/usr/lib/python3/dist-packages/setuptools/__init__.py", line 143, in setup
    return distutils.core.setup(**attrs)
  File "/usr/lib/python3.7/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/usr/lib/python3.7/distutils/dist.py", line 966, in run_commands
    self.run_command(cmd)
  File "/usr/lib/python3.7/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "/usr/lib/python3.7/distutils/command/build.py", line 135, in run
    self.run_command(cmd_name)
  File "/usr/lib/python3.7/distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/usr/lib/python3.7/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "setup.py", line 33, in run
    self.build_extension(ext)
  File "setup.py", line 58, in build_extension
    subprocess.check_call(['cmake', '--build', '.', '--target', 'install'], cwd=self.build_temp)
  File "/usr/lib/python3.7/subprocess.py", line 347, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['cmake', '--build', '.', '--target', 'install']' returned non-zero exit status 2.
```

I think the root of the problem is that `setup.py` now runs install in the `build_extension` stanza, via 6f720c56da:

```
        subprocess.check_call(['cmake', '--build', '.', '--target', 'install'], cwd=self.build_temp)
```

During build, `$DESTDIR` is not setup by packaging infrastructure so it tries to install into the root filesystem, not the staging one.

Not sure how install should work otherwise (I've never seen in handled specifically in `setup.py`), why was that line added in 1.3.1 and what's the purpose? If I remove it, gau2grid builds/installs just fine for me; this is what gets run (and which is mostly in line with `README.md`):

```
/usr/bin/python3 setup.py install --root `pwd`/debian/tmp
```

/cc @loriab

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -33,6 +33,7 @@ option_with_default(PYMOD_INSTALL_LIBDIR "Location within CMAKE_INSTALL_LIBDIR t
                                           Must start with: / . Used to imitate python install: /python3.6/site-packages ." /)
 option_with_print(INSTALL_PYMOD "Additionally installs as independent python module in PYMOD_INSTALL_LIBDIR" OFF)
 option_with_default(NATIVE_PYTHON_INSTALL "For INSTALL_PYMOD=ON, install in Python manner to PYTHON_EXECUTABLE's site-packages rather than Linux manner to prefix. Overrides CMAKE_INSTALL_PREFIX, CMAKE_INSTALL_LIBDIR, PYMOD_INSTALL_LIBDIR. Only Py module installed." OFF)
+option_with_print(NATIVE_PYTHON_INSTALL_WITH_LIB "Same as NATIVE_PYTHON_INSTALL except installs library, too, _without_ overriding CMAKE_INSTALL_* options." OFF)
 
 ########################  Process & Validate Options  ##########################
 include(autocmake_safeguards)
@@ -111,7 +112,7 @@ write_basic_package_version_file(${CMAKE_CURRENT_BINARY_DIR}/${PN}ConfigVersion.
                                  COMPATIBILITY AnyNewerVersion)
 
 # Install our files
-if(NOT(${INSTALL_PYMOD} AND ${NATIVE_PYTHON_INSTALL}))
+if(${NATIVE_PYTHON_INSTALL_WITH_LIB} OR (NOT(${INSTALL_PYMOD} AND ${NATIVE_PYTHON_INSTALL})))
     install(FILES ${CMAKE_CURRENT_BINARY_DIR}/gau2grid.h
                   ${CMAKE_CURRENT_BINARY_DIR}/gau2grid_pragma.h
                   DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/${PN})

--- a/setup.py
+++ b/setup.py
@@ -34,6 +34,7 @@ class CMakeBuild(build_ext):
 
     def build_extension(self, ext):
         global cmake_args
+        bypass_install = cmake_args.pop('-DBYPASS_INSTALL')
 
         internal_cmake_args = ['-DPYTHON_EXECUTABLE=' + sys.executable]
         internal_cmake_args += [k + "=" + v for k, v in cmake_args.items() if v]
@@ -55,7 +56,8 @@ class CMakeBuild(build_ext):
             os.makedirs(self.build_temp)
         subprocess.check_call(['cmake', ext.sourcedir] + internal_cmake_args, cwd=self.build_temp, env=env)
         subprocess.check_call(['cmake', '--build', '.'] + build_args, cwd=self.build_temp)
-        subprocess.check_call(['cmake', '--build', '.', '--target', 'install'], cwd=self.build_temp)
+        if not bypass_install:
+            subprocess.check_call(['cmake', '--build', '.', '--target', 'install'], cwd=self.build_temp)
 
 
 if __name__ == "__main__":
@@ -69,6 +71,7 @@ if __name__ == "__main__":
         '-DCMAKE_C_COMPILER': False,
         '-DCMAKE_PREFIX_PATH': False,
         '-DNATIVE_PYTHON_INSTALL_WITH_LIB': 'OFF',
+        '-DBYPASS_INSTALL': False,
     }
     invalid_args = {
         '-DBUILD_SHARED_LIBS': 'ON',

--- a/setup.py
+++ b/setup.py
@@ -68,6 +68,7 @@ if __name__ == "__main__":
         '-DCMAKE_C_FLAGS': False,
         '-DCMAKE_C_COMPILER': False,
         '-DCMAKE_PREFIX_PATH': False,
+        '-DNATIVE_PYTHON_INSTALL_WITH_LIB': 'OFF',
     }
     invalid_args = {
         '-DBUILD_SHARED_LIBS': 'ON',

